datasources/src/main/elm/sources/DataSourceForm.elm

```diff
@@ -0,0 +1,602 @@
+import Html.Events exposing (onCheck, onClick, onInput)
+import String.Extra
+import List.Extra
+import Messages exposing (..)
+import Utils exposing (nameToId)
+
+
+datasourceForm : Model -> DataSource -> Maybe DataSource -> Html Msg
+datasourceForm model datasource origin =
+  let
+
+    baseForm = [
+              div [ class "form-group" ] [
+                label [ for "name" ] [ text "Name" ]
+              , input [ type_ "text", class "form-control", id "name", value datasource.name, name "datasourceName", onInput (\s -> UpdateDataSource { datasource | name = s}) ] [] --, required ng-model "selectedDatasource.name", ng-change "updateKeyName(selectedDatasource.name)" ] [
+              , nameError
+              ]
+            , div [ class "form-group" ] [
+                label [ for "key"] [ text "Key name" ]
+              , idWarning
+              , input [ type_ "text", class "form-control has-example-help", id "key", placeholder (nameToId datasource.name), name "datasourceKey", value datasource.id, disabled (not isNew), onInput (\s -> UpdateDataSource { datasource | id = s}) ] []
+              , div [ class "example-help" ] [
+                  text "The key name will be used to access imported data, such as "
+                , pre [] [ text ("${node.properties["++ (if String.isEmpty datasource.id then "\"key_name\"" else  datasource.id )++"]") ]
+                ]
+              , span [ class "text-danger" ] [ idError ]
+              ]
+            , div [ class "form-group" ] [
+                label [ for "description", class "optional"] [ text "Description" ]
+              , textarea [ class "form-control", id "description", value datasource.description , onInput (\s -> UpdateDataSource { datasource | description = s})] []
+              ]
+            , div [ class "form-group" ] [
+                label [ class "bloc"] [ text "Type" ]
+              , div [ class "btn-group" ] [
+                  label [ class "btn btn-default active disabled" ] [ text "HTTP" ]
+                ]
+              ]
+              ]
+    enabled = datasource.enabled
+    actionButtons =
+        if model.hasWriteRights then
+        [
+                        div [ class "btn-group" ] [
+                          if (isNew) then text ""
+                          else
+                          button [ type_ "button", class "btn btn-danger" ] [--, ng-click "deleteDatasource()", ng-if "!selectedDatasource.isNew",
+                            text "Delete "
+                          , i [ class "fa fa-times-circle" ] []
+                          ]
+                        ]
+                      , div [ class "btn-group" ] [
+                          if (isNew) then text ""
+                          else
+                          button [ type_ "button", class "btn btn-default", onClick (SaveCall { datasource | enabled = not enabled } ) ] [
+                            text (if datasource.enabled then "Enable" else "Disable")
```

Comment:
it should be the contrary 
![image](https://github.com/Normation/rudder-plugins/assets/44649/6f17ed26-d3fc-4abe-abe3-f6fc9ae0cbfd)